collectors/ebpf.plugin/ebpf_hardirq.c

```diff
@@ -0,0 +1,503 @@
+            // when they're added, only then will we AVL insert.
+            v = callocz(1, sizeof(hardirq_val_t));
+            v->irq = key.irq;
+            v->dim_exists = false;
+
+            v_is_new = true;
+        }
+
+        // note two things:
+        // 1. we must add up latency value for this IRQ across all CPUs.
+        // 2. the name is unfortunately *not* available on all CPU maps - only
+        //    a single map contains the name, so we must find it. we only need
+        //    to copy it though if the IRQ is new for us.
+        bool name_saved = false;
+        uint64_t total_latency = 0;
+        int i;
+        int end = (running_on_kernel < NETDATA_KERNEL_V4_15) ? 1 : ebpf_nprocs;
+        for (i = 0; i < end; i++) {
+            total_latency += hardirq_ebpf_vals[i].latency/1000;
+
+            // copy name for new IRQs.
+            if (v_is_new && !name_saved && hardirq_ebpf_vals[i].name[0] != '\0') {
+                strncpyz(
+                    v->name,
+                    hardirq_ebpf_vals[i].name,
+                    NETDATA_HARDIRQ_NAME_LEN
+                );
+                name_saved = true;
+            }
+        }
+
+        // can now safely publish latency for existing IRQs.
+        v->latency = total_latency;
+
+        // can now safely publish new IRQ.
+        if (v_is_new) {
+            avl_t *check = avl_insert_lock(&hardirq_pub, (avl_t *)v);
+            if (check != (avl_t *)v) {
+                error("Internal error, cannot insert the AVL tree.");
+            }
+        }
+
+        key = next_key;
+    }
+}
+
+static void hardirq_read_latency_static_map(int mapfd)
+{
+    uint32_t i;
+    for (i = 0; i < HARDIRQ_EBPF_STATIC_END; i++) {
+        uint32_t map_i = hardirq_static_vals[i].idx;
+        int test = bpf_map_lookup_elem(mapfd, &map_i, hardirq_ebpf_static_vals);
+        if (unlikely(test < 0)) {
+            continue;
+        }
+
+        uint64_t total_latency = 0;
+        int cpu_i;
+        int end = (running_on_kernel < NETDATA_KERNEL_V4_15) ? 1 : ebpf_nprocs;
+        for (cpu_i = 0; cpu_i < end; cpu_i++) {
+            total_latency += hardirq_ebpf_static_vals[cpu_i].latency/1000;
+        }
+
+        hardirq_static_vals[i].latency = total_latency;
+    }
+}
+
+/**
+ * Read eBPF maps for hard IRQ.
+ */
+static void *hardirq_reader(void *ptr)
+{
+    read_thread_closed = 0;
+
+    heartbeat_t hb;
+    heartbeat_init(&hb);
+
+    ebpf_module_t *em = (ebpf_module_t *)ptr;
+
+    usec_t step = NETDATA_HARDIRQ_SLEEP_MS * em->update_time;
+    while (!close_ebpf_plugin) {
+        usec_t dt = heartbeat_next(&hb, step);
+        UNUSED(dt);
+
+        hardirq_read_latency_map(hardirq_maps[HARDIRQ_MAP_LATENCY].map_fd);
+        hardirq_read_latency_static_map(hardirq_maps[HARDIRQ_MAP_LATENCY_STATIC].map_fd);
+    }
+
+    read_thread_closed = 1;
+    return NULL;
+}
+
+static void hardirq_create_charts()
+{
+    ebpf_create_chart(
+        "system",
```

Comment:
Instead to use `"system"` here, I suggest you to use this [defintion](https://github.com/netdata/netdata/blob/f2e95f70819a70c6cffacb42784285f0e54e0bb4/collectors/ebpf.plugin/ebpf.h#L106), because  whether new menu comes with a different name for everything grouped inside the current `system`, this will simplify our life when we need to do modifications.

Reply:
👍